### ISSUE TITLE
OptGPSampler provide invalid samples with Gurobi

### ISSUE THREAD

**Issue by sulheim** [opencobra/cobrapy#807]
#### Problem description
I'm trying to generate random samples using the OptGPSampler. 
However, when I use Gurobi, all samples are invalid (outside solution space). When I use GLPK or CPLEX it works fine.

#### Code Sample
Create a [minimal, complete, verifiable example](https://stackoverflow.com/help/mcve).

```python
from cobra.test import create_test_model
from cobra.flux_analysis.sampling import OptGPSampler

model = create_test_model("textbook")

# Set solver to GLPK
model.solver = "glpk"

# Run random sampling
optpg = OptGPSampler(model, processes = 1)
s = optpg.sample(100)
print((optpg.validate(s) == "v").sum())

# With Gurobi
model.solver = "gurobi"
optpg = OptGPSampler(model, processes = 1)
s = optpg.sample(100)
print((optpg.validate(s) == "v").sum())
```

#### Actual Output
100 # Number of valid samples are 100 with GLPK
0     # and 0 with Gurobi

#### Expected Output
100
100

#### Dependency Information

Please paste the output of `python -c "from cobra import show_versions;
show_versions()"` in the details block below.

<details>

System Information
==================
OS         Windows
OS-release      10
Python       3.5.2

Package Versions
================
cobra        0.14.2
depinfo       1.4.0
future       0.17.1
numpy        1.15.4
optlang       1.4.3
pandas       0.24.0
pip          19.0.1
ruamel.yaml 0.15.87
setuptools   40.7.1
six          1.12.0
swiglpk       1.4.4
tabulate      0.8.1
wheel        0.29.0

</details>

**Comment by cdiener** [comment]
Hard for me to debug since I don't have access to Gurobi. But I do have some suspicions.

First of all could you update optlang to 1.4.4 and check whether the error persists? There were quite a few Gurobi fixes in that version...

**Comment by sulheim** [comment]
Updating optlang didn't solve it

**Comment by Midnighter** [comment]
I can test Gurobi aspects. It's a bit slower but if you let me know your hunches, I can try to investigate.

**Comment by cdiener** [comment]
Okay. Provided a fix in #808. This was a fun one. Didn't expect optlang would preserve order in some interfaces but not others.
